arangod/Cluster/ClusterComm.h

```diff
@@ -288,6 +293,15 @@ struct ClusterCommResult {
     // result was available in different status code situations :S
     if (single) {
       status = result->wasHttpError() ? CL_COMM_ERROR : CL_COMM_SENT;
+      if (status == CL_COMM_ERROR) {
+        try {
+          auto body = result->getBodyVelocyPack(VPackOptions());
+          if (body->slice().isObject() && body->slice().hasKey("errorMessage")) {
+            errorMessage = body->slice().get("errorMessage").copyString();
+            errorCode = body->slice().get("errorNum").getNumber<int>();
+          }
+        } catch (...) {}
```

Comment:
Or set some error code here.

Reply:
If the errorCode is still TRI_ERROR_NO_ERROR, look at what `getErrorCode()` does.

Reply:
Shouldn't it be a comm error instead of an internal error?

Reply:
CL_COMM_ERROR means the other side has returned some HTTP status code >= 400.
In this case we try to read it from the response. This may fail. Only in case we have a valid response and can read it, we read the details and store them in instance variables.
The errors are retrieved via getErrorCode, and there is still a check whether the errorCode was actually set or not. In case it was not set (by whatever reason), getErrorCode will fall back to its old behavior and return TRI_ERROR_INTERNAL.